Tasks/Common/sonarqube-common/sonarqube-common.ts

```diff
@@ -138,4 +171,68 @@ function isPrBuild(): boolean {
 
 function isNullOrEmpty(str) {
     return str === undefined || str === null || str.length === 0;
+}
+
+// Creates the string that comprises the build summary text, given the location of the /sonar build folder.
+function createSonarQubeBuildSummary(sqBuildFolder: string): string {
+    var taskReport: TaskReport = getSonarQubeTaskReport(sqBuildFolder);
+    if (!taskReport) {
+        throw new Error(tl.loc('sqAnalysis_TaskReportInvalid'));
+    }
+
+    return util.format('[%s >](%s "%s Dashboard")',
+        tl.loc('sqAnalysisBuildSummaryLine_OneProject'), taskReport.dashboardUrl, taskReport.projectKey);
+}
+
+// Returns the location of the SonarQube integration staging directory.
+export /* public for test purposes */ function getSonarQubeStagingDirectory(): string {
+    var sqStagingDir = path.join(tl.getVariable('build.artifactStagingDirectory'), ".sqAnalysis");
+    tl.mkdirP(sqStagingDir);
+    return sqStagingDir;
+}
+
+// Returns, as an object, the contents of the 'report-task.txt' file created by SonarQube plugins
+// The returned object contains the following properties:
+//   projectKey, serverUrl, dashboardUrl, ceTaskId, ceTaskUrl
+function getSonarQubeTaskReport(sonarPluginFolder: string): TaskReport {
+    var reportFilePath:string = path.join(sonarPluginFolder, 'report-task.txt');
+    if (!tl.exist(reportFilePath)) {
+        tl.debug('Task report not found at: ' + reportFilePath);
+        return null;
+    }
+
+    return createTaskReportFromFile(reportFilePath);
+}
+
+// Constructs a map out of an existing report-task.txt file. File must exist on disk.
+function createTaskReportFromFile(taskReportFile: string): TaskReport {
+    var reportFileString: string = fs.readFileSync(taskReportFile, 'utf-8');
+    if (!reportFileString) {
+        return null;
+    }
+
+    var reportLines: string[] = reportFileString.replace(/\r\n/g, '\n').split('\n'); // proofs against xplat line-ending issues
+
+    var reportMap = new Map<string, string>();
+    reportLines.forEach((reportLine:string) => {
+        var splitLine: string[] = reportLine.split('=');
+        if (splitLine.length > 1) {
+            reportMap.set(splitLine[0], splitLine.slice(1, splitLine.length).join());
+        }
+    });
+
+    var result:TaskReport;
+    try {
+        return result = TaskReport.createTaskReportFromMap(reportMap);
+    } catch (err) {
+        tl.debug(err.message);
+        throw new Error(tl.loc('sqAnalysis_TaskReportInvalid'));
```

Comment:
You seem to be mixing the 2 ways of erroring - throwing exception and returning null. Stick to one - I'd suggest you throw instead of returning null.